### PR TITLE
Changing Access to Ship weapons and Fixing Access for Cyborgs and AI shell

### PR DESCRIPTION
**Pull request by Sindorman**
While working on PR #15157 I noticed that hangar techs and miners have access to QM specific things. This included Tech Storage, Secure Ammunitions Storage and also QM's office. As I understand that is apparently to allow them to access mainly Secure Ammunition's storage and Flak Weapon. So in this PR:

- Added Ship weapons specific access type. Added such access to hanger technician, all head of staff, engineers.
- Added ship weapons access to mulebot
- Changed Secure Ammunition's storage, longbow, and flak gun door's access to ship weapons only. Also for Ammunition's storage bolt and blast door controls.
- Fixed access for the door that is between AI upload lobby and research hallway so that cyborgs can use it.
- Fixed access for AI shell to have all station access.

### CODE REVIEW

**Comment by Triogenix** [comment]
Bridge Crew shouldn't get access to the Grauwolf and Secure Ammunition Storage. It was very specifically removed previously from my understanding.

**Comment by Sindorman** [comment]
> 

Hmmm. I would need to confirm that with developers as I just mirrored access I saw already in code which seems like they did have access to at least Grauwolf, but not secure ammunition's storage

**Comment by Sindorman** [comment]
!review

**Comment by NonQueueingMatt** [comment]
> > Bridge Crew shouldn't get access to the Grauwolf and Secure Ammunition Storage. It was very specifically removed previously from my understanding. Then again I could be wrong.
> 
> Hmmm. I would need to confirm that with developers as I just mirrored access I saw already in code which seems like they did have access to at least Grauwolf, but not secure ammunition's storage

They should not have access to the guns themselves.

File: code/game/jobs/access_datum.dm [hunk 1]
```diff
@@ -614,4 +614,10 @@ var/const/access_kataphract_knight = 114
 /var/const/access_pra = 209
 /datum/access/access_pra
 	id = access_pra
-	access_type = ACCESS_TYPE_CENTCOM
\ No newline at end of file
+	access_type = ACCESS_TYPE_CENTCOM
+
+/var/const/access_ship_weapons = 210
```
Comment (SleepyGemmy) [inline code/game/jobs/access_datum.dm:619]: use a free ID from 1-99 since its on-ship access.

File: code/game/jobs/job/medical.dm [hunk 2]
```diff
@@ -274,8 +274,8 @@
 		SPECIES_SKRELL_AXIORI = 55
 	)
 
-	access = list(access_medical, access_medical_equip, access_morgue, access_surgery, access_pharmacy, access_virology, access_eva, access_maint_tunnels, access_engine, access_research, access_sec_doors, access_external_airlocks, access_psychiatrist, access_first_responder)
-	minimal_access = list(access_medical, access_medical_equip, access_morgue, access_surgery, access_eva, access_maint_tunnels, access_engine, access_research, access_sec_doors, access_external_airlocks, access_first_responder)
+	access = list(access_medical, access_medical_equip, access_morgue, access_surgery, access_pharmacy, access_virology, access_eva, access_maint_tunnels, access_engine, access_ship_weapons, access_research, access_sec_doors, access_external_airlocks, access_psychiatrist, access_first_responder)
+	minimal_access = list(access_medical, access_medical_equip, access_morgue, access_surgery, access_eva, access_maint_tunnels, access_engine, access_ship_weapons, access_research, access_sec_doors, access_external_airlocks, access_first_responder)
```
Comment (SleepyGemmy) [inline code/game/jobs/job/medical.dm:278]: i don't see why first responders would have access to the ship's weapons.

File: code/game/jobs/job/outsider/representative.dm [hunk 3]
```diff
@@ -16,7 +16,7 @@
 		SPECIES_SKRELL_AXIORI = 80
 	)
 
-	access = list(access_lawyer, access_maint_tunnels)
+	access = list(access_lawyer, access_maint_tunnels, access_ship_weapons)
```
Comment (SleepyGemmy) [inline code/game/jobs/job/outsider/representative.dm:19]: liaisons should definitely not have ship weapons access.

File: code/game/machinery/bots/mulebot.dm [hunk 4]
```diff
@@ -60,7 +60,7 @@
 	. = ..()
 	wires = new(src)
 	botcard = new(src)
-	botcard.access = list(access_maint_tunnels, access_mailsorting, access_cargo, access_cargo_bot, access_qm, access_mining, access_mining_station)
+	botcard.access = list(access_maint_tunnels, access_mailsorting, access_cargo, access_cargo_bot, access_ship_weapons, access_qm, access_mining, access_mining_station)
```
Comment (SleepyGemmy) [inline code/game/machinery/bots/mulebot.dm:63]: mulebots are deprecated as far as i know, but technically fine.

File: html/changelogs/Sindorman-access.yml [hunk 5]
```diff
@@ -0,0 +1,44 @@
...
+# Optional: Remove this file after generating master changelog.  Useful for PR changelogs that won't get used again.
+delete-after: True
+
+# Any changes you've made.  See valid prefix list above.
+# INDENT WITH TWO SPACES.  NOT TABS.  SPACES.
+# SCREW THIS UP AND IT WON'T WORK.
+# Also, all entries are changed into a single [] after a master changelog generation. Just remove the brackets when you add new entries.
+# Please surround your changes in  double quotes ("), as certain characters otherwise screws up compiling. The quotes will not show up in the changelog.
+changes:
+  - tweak: "Added Ship Weapons access type for flak cannon and Secure Ammunitions Storage. Roles that now have it: hangar technicia, all head of staff, bridge crew, engineers, rescue tech, lawyer, HR assistant."
```
Comment (SleepyGemmy) [inline html/changelogs/Sindorman-access.yml:41]: spellcheck and look over the roles.

File: maps/sccv_horizon/sccv_horizon-2_deck_2.dmm [hunk 6]
```diff
@@ -9072,7 +9072,7 @@
 	},
 /obj/machinery/door/airlock/multi_tile/glass{
 	name = "Research Hallway";
-	req_access = list(47)
+	req_access = list(16,47)
```
Comment (SleepyGemmy) [inline maps/sccv_horizon/sccv_horizon-2_deck_2.dmm:9075]: this should be `req_one_access`.

File: maps/sccv_horizon/sccv_horizon-2_deck_2.dmm [hunk 7]
```diff
@@ -44107,7 +44107,7 @@
 	input_tag = "cooling_in";
 	name = "Engine Cooling Control";
 	output_tag = "cooling_out";
-	sensors = list("engine_sensor"="Engine Core")
+	sensors = list("engine_sensor" = "Engine Core")
```
Comment (SleepyGemmy) [inline maps/sccv_horizon/sccv_horizon-2_deck_2.dmm:44110]: ?

**Review by SleepyGemmy — CHANGES_REQUESTED** [review]
image
don't change back the floor tiles to plasteel.

**Comment by Sindorman** [comment]
> image don't change back the floor tiles to plasteel.

I did not change any tiles, look at the map diff. There are no changes to the tiles anywhere

**Comment by Sindorman** [comment]
!review

File: code/game/jobs/job/captain.dm [hunk 8]
```diff
@@ -171,7 +171,7 @@ var/datum/announcement/minor/captain_announcement = new(do_newscast = 1)
 	outfit = /datum/outfit/job/bridge_crew
 
 	access = list(access_eva, access_heads, access_maint_tunnels, access_weapons, access_bridge_crew, access_intrepid, access_teleporter)
-	minimal_access = list(access_heads, access_eva, access_gateway, access_weapons, access_bridge_crew, access_intrepid, access_teleporter)
+	minimal_access = list(access_heads, access_eva, access_gateway, access_weapons, access_ship_weapons, access_intrepid, access_teleporter)
```
Comment (NonQueueingMatt) [inline code/game/jobs/job/captain.dm:174]: Bridge crew shouldn't have access.

**Review by NonQueueingMatt — CHANGES_REQUESTED** [review]

File: code/game/jobs/job/captain.dm [hunk 9]
```diff
@@ -171,7 +171,7 @@ var/datum/announcement/minor/captain_announcement = new(do_newscast = 1)
 	outfit = /datum/outfit/job/bridge_crew
 
 	access = list(access_eva, access_heads, access_maint_tunnels, access_weapons, access_bridge_crew, access_intrepid, access_teleporter)
-	minimal_access = list(access_heads, access_eva, access_gateway, access_weapons, access_bridge_crew, access_intrepid, access_teleporter)
```
Comment (SleepyGemmy) [inline code/game/jobs/job/captain.dm:174]: restore bridge crew access to bridge crew.

**Review by SleepyGemmy — CHANGES_REQUESTED** [review]

File: code/game/jobs/job/captain.dm (hunk 8 above)
Comment (Sindorman) [reply code/game/jobs/job/captain.dm:174]: Ah, I removed the wrong access lol

**Comment by Sindorman** [comment]
!review

**Comment by Sindorman** [comment]
Common github, it does not let me re-request review from both lol

**Review by SleepyGemmy — APPROVED** [review]

**Review by JohnWildkins — APPROVED** [review]

**Pull request merged**